geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/GeoMesaMetadataStats.scala

```diff
@@ -49,13 +50,24 @@ class GeoMesaMetadataStats(val ds: AccumuloDataStore, statsTable: String, genera
   private val compactionScheduled = new AtomicBoolean(false)
   private val lastCompaction = new AtomicLong(0L)
 
-  executor.scheduleWithFixedDelay(new Runnable() {
-    override def run(): Unit =
+  private val running = new AtomicBoolean(true)
+  private var scheduledCompaction: ScheduledFuture[_] = null
+
+  private val compactor = new Runnable() {
+    override def run(): Unit = {
       if (lastCompaction.get < DateTime.now(DateTimeZone.UTC).minusHours(1).getMillis &&
           compactionScheduled.compareAndSet(true, false) ) {
         compact()
       }
-  }, 1, 1, TimeUnit.HOURS)
+      if (running.get) {
+        scheduledCompaction = executor.schedule(this, 1, TimeUnit.HOURS)
+      } else {
+        scheduledCompaction = null
```

Comment:
Does setting the scheduledCompaction to null stop/cancel the previously scheduled compactions?